src/LibraryManager/Manifest.cs

```diff
@@ -228,16 +229,34 @@ private int DeleteLibraryFiles(ILibraryInstallationState state, Action<string> d
         {
             int filesDeleted = 0;
 
-            foreach (string file in state.Files)
+            IProvider provider = _dependencies.GetProvider(state.ProviderId);
+            ILibraryInstallationResult updatedStateResult = provider.UpdateStateAsync(state, new CancellationToken()).Result;
```

Comment:
`new CancellationToken()` is not great.  If you want to be able to cancel the operation, you need a `CancellationTokenSource`.  If you don't, then you should use `CancellationToken.None` to indicate the intent.